src/aws-cpp-sdk-core/include/smithy/client/AwsSmithyClient.h

```diff
@@ -70,28 +67,32 @@ namespace client
             assert(m_authSchemeResolver);
```

Comment:
shouldn't assert but handle gracefully with error outcome. Unless the asserts are stripped out in the final build this causes bad API user experience IMO